### PR TITLE
WIP: name and ip selector option

### PR DESCRIPTION
**Pull request by HarleyKwyn**

### CODE REVIEW

**Comment by HarleyKwyn** [comment]
#53

File: lib/controller.js [hunk 1]
```diff
@@ -50,37 +59,83 @@ function listTessels(options) {
...
+      if(!collector.default){
+        collector.default = connection;
+      }
+      // Return with default and selected, if any
+      return collector;
+    }, {default: null, selected: null} )
+    .then(function(connections){
+      // Check if one was selected, if not, use default;
+      var connection = connections.selected ? connections.selected : connections.default;
+      var message = "Connected over" + connection.connectionType + "."
```
Comment (johnnyman727) [inline lib/controller.js]: You need to add a space between "Connected over" and `connection.connectionType`.

**Comment by johnnyman727** [comment]
What happens if you provide a flag with a name that doesn't exist?

**Comment by johnnyman727** [comment]
With command `t2 run test/test-deploy-script.js -n t2.local`

```
Unhandled rejection TypeError: Cannot read property 'host' of undefined
    at Promise.settle.reduce.then.then.reduce (/Users/Jon/Work/technical/t2-cli/lib/controller.js:97:50)
    at tryCatcher (/Users/Jon/Work/technical/t2-cli/node_modules/bluebird/js/main/util.js:24:31)
    at ReductionPromiseArray._promiseFulfilled (/Users/Jon/Work/technical/t2-cli/node_modules/bluebird/js/main/reduce.js:107:18)
    at ReductionPromiseArray.init [as _init$] (/Users/Jon/Work/technical/t2-cli/node_modules/bluebird/js/main/promise_array.js:92:18)
    at Promise._settlePromiseAt (/Users/Jon/Work/technical/t2-cli/node_modules/bluebird/js/main/promise.js:528:21)
    at Promise._settlePromises (/Users/Jon/Work/technical/t2-cli/node_modules/bluebird/js/main/promise.js:646:14)
    at Async._drainQueue (/Users/Jon/Work/technical/t2-cli/node_modules/bluebird/js/main/async.js:79:16)
    at Async._drainQueues (/Users/Jon/Work/technical/t2-cli/node_modules/bluebird/js/main/async.js:89:10)
    at Async.drainQueues (/Users/Jon/Work/technical/t2-cli/node_modules/bluebird/js/main/async.js:14:14)
    at process._tickCallback (node.js:355:11)
```

**Pull request merged**